subprojects/build-cache/src/main/java/org/gradle/caching/local/internal/H2BuildCacheService.java

```diff
@@ -81,8 +100,13 @@ public boolean contains(BuildCacheKey key) {
     @Override
     public boolean load(BuildCacheKey key, BuildCacheEntryReader reader) throws BuildCacheException {
         try (Connection conn = dataSource.getConnection()) {
-            try (PreparedStatement stmt = conn.prepareStatement("select entry_content from filestore.catalog where entry_key = ?")) {
+            try (PreparedStatement stmt = conn.prepareStatement(
+                "select entry_content from filestore.catalog where entry_key = ?;" +
+                    "update filestore.lru set entry_accessed = ? where entry_key = ?;"
+            )) {
                 stmt.setString(1, key.getHashCode());
+                stmt.setLong(2, clock.getCurrentTime());
```

Comment:
Yeah, good catch... I think this should not be an issue, since we anyway expect that some entries do no exist for a manifest, since manifests can share entries. So some entries for a manifest can be accessed also when reading some other manifest.
  
But we could still improve that, but we need to change the protocol interface.